### PR TITLE
fix(autonomous): reconcile settlement amounts with the live wallet balance

### LINKED ISSUE #201: bug: exit settlement amount never reconciled with wallet balance — concurrent entry consumed the USDC, swap sim fails 130x, job terminal, sweep can't recover: 4.35 (25% of equity) stranded
## Problem

An exit settlement's sell amount is never reconciled with the actual wallet balance at execution time. A concurrent entry consumed part of the wallet USDC that the pending settlement job still expected, so every swap attempt fails simulation ("Swap simulation failed" — insufficient balance), the job exhausts attempts and goes `terminal` (0.2.0 classifies it non-transient), and the orphan sweep **cannot recover it**: `backedMints` includes the tokens of open positions, and the two open HYPE/USDC positions make ALL wallet USDC "position-backed", even though position liquidity lives in the position account, not the wallet. Net result: **$24.35 (25% of live equity) permanently stranded with no recovery path**, and the agent stays capital-locked (free SOL 0.397 < 0.4463-0.4963 needed for any entry).

## Evidence (live instance, wallet 72JLRhFo3u1qxA6qnH6tduqtw1oj88PChf67WNFhHxds)

- Settlement job `046e364d` (exit of position `B7SxsLfc…`, SOL/USDC pool BGm1tav5): created 2026-08-08T11:52:34Z, `amount_atomic=41,913,604` USDC, destination SOL, **130 attempts**, status `terminal`, `error="Swap simulation failed"`, `updated_at=2026-08-09T05:42:56Z` (0.2.0 process's own final attempt). No tx signature.
- Wallet USDC balance: **24.35** (snapshot 2026-08-09T01:30:31Z; still 24.35 at 05:50Z). 41.91 - 24.35 = 17.56 USDC consumed by the HYPE/USDC entry `5VzUiusjS…` entered ~2026-08-08T12:06Z (14 min after the exit) — the entry funded its USDC side from the wallet's exit proceeds instead of reserving/swapping.
- Sweep (0.2.0, identical logic in 0.1.12): `backedMints` = mints with non-final jobs + `tokenX`/`tokenY` of every open position (fetched via `getPoolState`). Two open HYPE/USDC positions -> USDC in `backedMints` -> the 24.35 USDC holding is skipped every cycle. 0.2.0 has run 4+ cycles: no re-queue, jobs list unchanged (16, of which the 2 USDC terminal jobs).
- Entries: `[wallet-reserve] free SOL 0.3970 < needed 0.4963 — entry skipped (capacity-limited, not an error)` — all ENTERs skipped since 01:43Z; the stranded USDC never converts back to SOL.

## Root causes

1. **Stale settlement amount**: the job sells 41.91 USDC while the wallet holds 24.35 — quote succeeds, swap simulation fails (insufficient balance). The job amount is set once at creation and never clamped to the live wallet balance.
2. **Entry funding ignores pending settlement claims**: a new position's USDC side was funded from wallet USDC that a pending exit settlement was about to sell.
3. **Sweep blind spot (amount-unaware backing)**: `backedMints` conflates position-backed tokens with wallet-held balances of the same mint. Position liquidity is in the position account — wallet holdings of the same mint are still sweepable.

## Proposed fix

- Before quoting/swapping a settlement, clamp `amountAtomic` to `min(job.amountAtomic, wallet.amountAtomic)`; on insufficiency, re-queue with the actual balance instead of failing.
- Entry prep: reserve (or exclude from spendable) funds claimed by pending/active settlement jobs for the same mint.
- Sweep: make backing amount-aware — skip a mint only if the wallet holding does not exceed what open positions/reserved jobs require; terminal jobs with unspent balance should always win over position backing for the wallet-held excess.

### PR DESCRIPTION
**Pull request by irfndi**
Fixes #201

## Problem

An exit settlement's sell amount is fixed at creation and never reconciled with the live wallet balance. A concurrent entry consumed part of the wallet USDC the pending settlement still expected (41.91 USDC expected, 24.35 in wallet): every swap attempt failed simulation ("Swap simulation failed" — insufficient balance) 130 times, the job went terminal, and the orphan sweep could not recover it — `backedMints` includes open-position legs, and the two HYPE/USDC positions made ALL wallet USDC "position-backed" even though position liquidity lives in the position account. Result: $24.35 (25% of live equity) permanently stranded.

## Fix

1. **Settlement amount clamp** (`processSettlementJobs`): the sell amount is clamped to `min(job amount, live wallet balance)` before quoting. A partial balance persists through the retryable job (next attempt quotes what exists); a fully-consumed balance terminalizes with a clear error instead of looping; a failed balance read fails open; partial adapters fail open instead of defecting.
2. **Amount-aware sweep backing** (`sweepOrphanSettlements`): position backing no longer exempts a mint that has a stranded-terminal job — the terminal job's unspent balance proves the wallet-held excess was never sold, so the sweep revives it with the LIVE wallet amount. Active-job backing still suppresses (an in-flight job owns the mint).
3. **Entry prep reserves pending settlement claims** (`entry-prep-service`): the spendable balance subtracts non-final settlement job amounts per mint, so a new entry can no longer consume funds an exit settlement is about to sell. `EntryPrepLive` now takes `DbService` (wired through `buildLayer` and test layers).

## Verification

- New regression tests: processor clamps the quote amount to the wallet balance; zero-balance terminalizes without quoting; the sweep revives a terminal job despite position backing with the live wallet amount; entry prep rejects with `INSUFFICIENT_USDC_BALANCE` when a pending claim makes the USDC leg unaffordable.
- `bun run lint` clean; `bun run test` 123 files / 1610 tests pass.

## Summary by Sourcery

Reconcile autonomous exit settlements with the live wallet balance to prevent stranded funds and infinite retry loops.

Bug Fixes:
- Clamp settlement sell amounts to the current wallet balance when processing jobs, terminalizing settlements whose balances have been fully consumed instead of repeatedly failing swaps.
- Adjust orphan settlement sweeping so wallet-held excess for a mint is revived even when that mint is position-backed, while still avoiding double-selling mints with active jobs.
- Reserve pending non-final settlement job amounts from entry preparation spendable balances so new entries cannot consume funds that existing settlements expect to sell.

Enhancements:
- Wire database access into the entry preparation service and runtime to support settlement-aware balance calculations.
- Extend autonomous runtime and sweep logic with wallet and position awareness to better model live holdings at execution time.

Documentation:
- Document the settlement reconciliation behaviour and its safeguards in the unreleased CHANGELOG entry.

Tests:
- Add regression tests covering clamped quote amounts, terminalization on zero wallet balance, orphan sweep revival of terminal jobs with wallet-held excess, and entry preparation failures when pending settlement claims exhaust USDC balance.
- Update existing autonomous token and entry-prep tests to mock wallet balances and DB services required for the new reconciliation logic.

### CODE REVIEW

File: engine/entry-prep-service.ts [hunk 1]
```diff
@@ -302,23 +304,56 @@ export const EntryPrepLive = Layer.effect(
...
+                claim = BigInt(job.amountAtomic);
+              } catch {
+                continue;
+              }
+              if (claim <= 0n) continue;
+              pendingClaims.set(job.tokenMint, (pendingClaims.get(job.tokenMint) ?? 0n) + claim);
+            }
+          }
+          const claimedX = pendingClaims.get(pool.tokenX) ?? 0n;
+          const claimedY = pendingClaims.get(pool.tokenY) ?? 0n;
```
[the remark this answers is not in the document]
Comment (irfndi) [reply engine/entry-prep-service.ts:332]: Fixed: the USDC funding path now subtracts pending USDC settlement claims from the spendable balance — a token/token pool entry can no longer consume USDC an exit settlement is about to sell (the leg-only reserve missed this path). Regression test: no-claim swap proceeds, claimed USDC rejects with INSUFFICIENT_USDC_BALANCE.
